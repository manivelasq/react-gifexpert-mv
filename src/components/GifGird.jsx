import PropTypes from "prop-types";
import { GifItem } from './GifItem';
import { useFetchGifs } from '../hooks/useFetchGifs';

export const GifGird = ( { category} ) => {

    const { images, isLoading } = useFetchGifs(category);

    return (
        <>
            <h3>{category}</h3>
            {
                isLoading && (<h2>Cargando....</h2>)
            }
            
            <div className='card-grid'>
                {
                    images.map( (img) => (
                        <GifItem 
                            key= { img.id }
                            { ...img}
                        />
                    ))
                }
            </div>
        </>
    )

}

GifGird.propTypes = {
    category: PropTypes.string.isRequired,
}