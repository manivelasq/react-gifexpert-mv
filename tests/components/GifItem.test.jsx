import React from 'react'
import { render, screen } from '@testing-library/react';
import {expect, test} from '@jest/globals';
import { GifItem } from '../../src/components/GifItem';

/*Tarea

1. AÃ±adir los PropTypes
    a. Title obligatorio
    b. Url Obligatorio
2. Evaluar el snapshot
*/

describe('Pruebas en <GifItem />', () => { 

    const title = "Esta es una prueba";
    const id = "11";
    const url = "http://localhost/";

    test('debe hacer match con el Snapshot', () => {
        const { container } = render( 
                                        <GifItem 
                                            title = {title} 
                                            url = {url} 
                                        />
                                    );
        expect ( container ).toMatchSnapshot();
    })

    test('debe de mostrar la imagen con el URL y el ALT indicado', () => {

        render( 
            <GifItem 
                title = {title} 
                url = {url} 
            />
        );
        // screen.debug();
        // console.log(screen.getByRole('img').src);
        // console.log(screen.getByRole('img').alt);
        // expect(screen.getByRole('img').src).toBe(url);
        // expect(screen.getByRole('img').alt).toBe(title);

        const {src, alt} = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
        
    })

    test('debe de mostrar el titulo en el componente', () => {

        render( 
            <GifItem 
                title = {title} 
                url = {url} 
            />
        );

        // screen.debug();
        expect(screen.getByText(title)).toBeTruthy();

    })

});
