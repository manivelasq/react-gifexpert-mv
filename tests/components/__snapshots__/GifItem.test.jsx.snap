// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifItem /> debe hacer match con el Snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Esta es una prueba"
      src="http://localhost/"
    />
    <p>
      Esta es una prueba
    </p>
  </div>
</div>
`;
